src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForMultipleBuildingBlocks.cs

```diff
@@ -0,0 +1,134 @@
+               messageBuilder.AppendLine();
+
+               foreach (var item in referringSimulationsAndBuildingBlocks)
+               {
+                  var buildingBlock = item.Item2;
+                  var simulations = item.Item1;
+
+                  messageBuilder.AppendLine($"\t- {AppConstants.CannotRemoveBuildingBlockFromProject(buildingBlock.Name, simulations.AllNames())}");
+               }
+
+               _interactionTaskContext.DialogCreator.MessageBoxInfo(messageBuilder.ToString());
+            }
+         }
+
+         private void removeBuildingBlock(IBuildingBlock buildingBlockToRemove)
+         {
+            switch (buildingBlockToRemove)
+            {
+               case EventGroupBuildingBlock eventGroupBuildingBlock:
+                  _interactionTasksForApplicationMoleculeBuilder.Remove(eventGroupBuildingBlock, eventGroupBuildingBlock.Module, eventGroupBuildingBlock, true);
+                  break;
+
+               case InitialConditionsBuildingBlock initialConditionsBuildingBlock:
+                  _interactionTasksForInitialCondition.Remove(initialConditionsBuildingBlock, initialConditionsBuildingBlock.Module, initialConditionsBuildingBlock, true);
+                  break;
+
+               case MoleculeBuildingBlock moleculeBuildingBlock:
+                  _interactionTasksForMolecule.Remove(moleculeBuildingBlock, moleculeBuildingBlock.Module, moleculeBuildingBlock, true);
+                  break;
+
+               case PassiveTransportBuildingBlock passiveTransportBuildingBlock:
+                  _interactionTasksForPassiveTransport.Remove(passiveTransportBuildingBlock, passiveTransportBuildingBlock.Module, passiveTransportBuildingBlock, true);
+                  break;
+
+               case ParameterValuesBuildingBlock parameterValuesBuildingBlock:
+                  _interactionTasksForParameterValues.Remove(parameterValuesBuildingBlock, parameterValuesBuildingBlock.Module, parameterValuesBuildingBlock, true);
+                  break;
+
+               case MoBiReactionBuildingBlock mobiReactionBuildingBlock:
+                  _interactionTasksForMobiReaction.Remove(mobiReactionBuildingBlock, mobiReactionBuildingBlock.Module, mobiReactionBuildingBlock, true);
+                  break;
+
+               case MoBiSpatialStructure mobiSpatialStructure:
+                  _interactionTasksForMobiSpatialStructure.Remove(mobiSpatialStructure, mobiSpatialStructure.Module, mobiSpatialStructure, true);
+                  break;
+
+               case ObserverBuildingBlock observerBuildingBlock:
+                  _interactionTasksForObserver.Remove(observerBuildingBlock, observerBuildingBlock.Module, observerBuildingBlock, true);
+                  break;
+
+               default:
+                  throw new InvalidOperationException($"No interaction task found for building block type: {buildingBlockToRemove.GetType()}");
+            }
+         }
```

Comment:
You'll need to capture the returned commands from each of these and make sure they are added to the history.

Also, where you have this constant parameter `true`, use the parameter name so we know what it does `silent: true`